From ed82c63c43018dc8c1270f12f6d667bafd232e4d Mon Sep 17 00:00:00 2001
From: Peter Williams <peter@newton.cx>
Date: Tue, 11 Sep 2018 12:51:29 -0400
Subject: [PATCH 2/5] Disable some more Windows-unfriendly tests.

---
 test/Makefile.am | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/test/Makefile.am b/test/Makefile.am
index 56129d7..93382de 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -11,7 +11,7 @@ AM_TESTS_ENVIRONMENT= \
 BUILT_SOURCES = $(builddir)/out.expected
 
 SH_LOG_COMPILER = sh
-TESTS=run-test.sh
+TESTS=
 
 TESTDATA =			\
 	4x6.pcf			\
@@ -74,11 +74,6 @@ endif
 check_PROGRAMS += test-bz106618
 test_bz106618_LDADD = $(top_builddir)/src/libfontconfig.la
 
-check_PROGRAMS += test-hash
-test_hash_CFLAGS = -I$(top_builddir) -I$(top_builddir)/src $(UUID_CFLAGS)
-test_hash_LDADD = $(UUID_LIBS)
-TESTS += test-hash
-
 EXTRA_DIST=run-test.sh run-test-conf.sh $(TESTDATA) out.expected-long-family-names out.expected-no-long-family-names
 
 CLEANFILES=out out1 out2 fonts.conf out.expected
-- 
2.17.1

