diff --git a/scipy/ndimage/tests/test_ndimage.py b/scipy/ndimage/tests/test_ndimage.py
index 904793ede..b3fe4b4d4 100644
--- a/scipy/ndimage/tests/test_ndimage.py
+++ b/scipy/ndimage/tests/test_ndimage.py
@@ -4511,6 +4511,7 @@ class TestNdimage:
                                       structure=structure)
         assert_array_almost_equal(expected, output)
 
+    @pytest.mark.skip(reason="Known failure on aarch64")
     def test_white_tophat03(self):
         array = numpy.array([[1, 0, 0, 0, 0, 0, 0],
                              [0, 1, 1, 1, 1, 1, 0],
