From e612970e267a599dc2c94b3a3bf4390632ea8134 Mon Sep 17 00:00:00 2001
From: Peter Williams <peter@newton.cx>
Date: Mon, 3 Sep 2018 23:20:56 -0400
Subject: [PATCH 2/2] Increase some test timeouts.

---

diff --git a/tests/refcount/meson.build b/tests/refcount/meson.build
index 68c23cc..71d9bcb 100644
--- a/tests/refcount/meson.build
+++ b/tests/refcount/meson.build
@@ -5,7 +5,7 @@ refcount_tests = {
   'properties' : {},
   'properties2' : {'suite' : ['slow']},
   'properties3' : {'suite' : ['slow']},
-  'properties4' : {},
+  'properties4' : {'suite' : ['slow']},
   'signal1' : {
     'source' : 'signals.c',
     'c_args' : ['-DTESTNUM=1'],
