From 7d8946821dcaa3e2023ee386ebf9daacce102ce1 Mon Sep 17 00:00:00 2001
From: Mike Sarahan <msarahan@gmail.com>
Date: Mon, 2 Apr 2018 16:18:26 +0000
Subject: [PATCH] disable fd-limit tests (not supported in docker)

---
 test/dbus-daemon.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/test/dbus-daemon.c b/test/dbus-daemon.c
index 222b102..df13b09 100644
--- a/test/dbus-daemon.c
+++ b/test/dbus-daemon.c
@@ -2121,10 +2121,10 @@ main (int argc,
   g_test_add ("/unix-runtime-is-default", Fixture, &listen_unix_runtime_config,
       setup, test_echo, teardown);
 
-  g_test_add ("/fd-limit/session", Fixture, NULL,
-              setup, test_fd_limit, teardown);
-  g_test_add ("/fd-limit/system", Fixture, &as_another_user_config,
-              setup, test_fd_limit, teardown);
+  // g_test_add ("/fd-limit/session", Fixture, NULL,
+  //             setup, test_fd_limit, teardown);
+  // g_test_add ("/fd-limit/system", Fixture, &as_another_user_config,
+  //             setup, test_fd_limit, teardown);
 #endif
 
   return g_test_run ();
-- 
2.16.1

